opus/application/static_media/js/browse.js

```diff
@@ -613,23 +620,61 @@ var o_browse = {
         $("#op-observation-number").html(value);
     },
 
+    // This function will be called when we scroll the slide to a target value
     onUpdateSlider: function(value) {
+        let tab = `#${opus.prefs.view}`;
         let elem = $(`#${opus.prefs.view} .thumbnail-container[data-obs="${value}"]`);
+
         if (elem.length > 0) {
             o_browse.setScrollbarOnSlide(value);
+            // Update obsNum in infiniteScroll instances, and this obsNum is the first item in current page. (will be used to set scrollbar position in renderGalleryAndTable). This is for the case when above o_browse.setScrollbarOnSlide(value) trigger infiniteScroll load event.
+            $(`${tab} .op-gallery-view`).infiniteScroll({"obsNum": value});
+            $(`${tab} .op-dataTable-view`).infiniteScroll({"obsNum": value});
         } else {
-            o_browse.loadData(value);
+            // when scrolling on slider and loadData is called, we will fetch 3 * getLimit items (one current page, one next page, and one previous page).
+            // firstObs will be the very first obsNum for data rendering this time
+            let firstObs = Math.max(value - o_browse.getLimit(), 1);
+
+            // if value + 1 * getLimit is larger than result counts, we will prefetch more data ahead of firstObs
+            if ((value + o_browse.getLimit() - 1) > opus.resultCount) {
```

Comment:
I don't understand the purpose of this code. If we aren't going to have enough data to prefetch, what's the purpose of prefetching extra data above the requested obs? We could just prefetch less data and still have the (most of) current and entire previous page available. Also, as written, if there requested page runs even 1 obs over the result count limit, we prefetch an *entire* extra previous page, which seems wasteful.